### PR TITLE
Catch all exception when consume kafka record

### PR DESCRIPTION
**Pull request by nileblack**
### Fix "Illegal Kafka record broke ConsumerThread"

KafkaFetcher Consume Record from kafka, but when One Record IS BAD, data has some error, or even just occur some error when process the data, it will terminate the  "ConsumerThread", NO data can processed anymore.

There is a Dirty & Quick fix, try catch all exception.

### CODE REVIEW

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java [hunk 1]
```diff
@@ -137,7 +137,11 @@ public void run() {
                 Iterator<ConsumerRecord<String, Bytes>> iterator = consumerRecords.iterator();
                 while (iterator.hasNext()) {
                     ConsumerRecord<String, Bytes> record = iterator.next();
-                    handlerMap.get(record.topic()).handle(record);
+                    try {
+                        handlerMap.get(record.topic()).handle(record);
+                    } catch(Throwable t) {
+                        log.error("consume record error", t);
```
Comment (wu-sheng) [inline oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java:143]: When and why this exception happens?

**Comment by wu-sheng** [comment]
2 thing, make sure your code format is correct. and update the change log file, as PR template said.

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java (hunk 1 above)
Comment (nileblack) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java:143]: aha ,the log is lost. In my situation , `ServiceManagementHandler.java`  `L65` throw a NPE.
```
    @Override
    public void handle(final ConsumerRecord<String, Bytes> record) {
        try {
            if (record.key().startsWith("register-")) {
                serviceReportProperties(InstanceProperties.parseFrom(record.value().get()));
            } else {
                keepAlive(InstancePingPkg.parseFrom(record.value().get()));
            }
        } catch (InvalidProtocolBufferException e) {
            log.error("", e);
        }
    }
```
Maybe a record without a key?

Comment (wu-sheng) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java:143]: I am fine if you just want to make sure the illegal data would not break all things. 
At the same time, fix doc and ci.

**Comment by wu-sheng** [comment]
Please update CHANGES.md

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java [hunk 2]
```diff
@@ -132,14 +132,23 @@ public void start() {
...
+                ConsumerRecords<String, Bytes> consumerRecords = consumer.poll(Duration.ofMillis(500L));
+                if (!consumerRecords.isEmpty()) {
+                    Iterator<ConsumerRecord<String, Bytes>> iterator = consumerRecords.iterator();
+                    while (iterator.hasNext()) {
+                        ConsumerRecord<String, Bytes> record = iterator.next();
+                        try {
+                            handlerMap.get(record.topic()).handle(record);
+                        } catch (Throwable e) {
+                            log.error("process record error, topic:{}, key:{}, value:{}", record.topic(), record.key(), record.value());
+                            log.error("process record error", e);
```
Comment (wu-sheng) [inline oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java:145]: Why need two error lines?

Comment (dmsolr) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java:145]: Once a record is failed to handle, a portion records of the batch will be discarded.

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java (hunk 1 above)
Comment (dmsolr) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java]: As you show the code, I suggest changing the `InvalidProtocolBufferException` to `Throwable` or `Exception`. And then don't need to add try-catch statement again.

Comment (nileblack) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java]: There mayby Another `Handler`  throws a uncaputure exception.

Comment (dmsolr) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java]: Yeap, I mean all handlers should be modified. :)

**Comment by dmsolr** [comment]
Notice that, #5718 was merged, please take into account how to do this.

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java (hunk 1 above)
Comment (zifeihan) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java]: i think every hander should catch `Exception` not `InvalidProtocolBufferException`.

**Comment by wu-sheng** [comment]
@nileblack Are you still working on this?

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java (hunk 2 above)
Comment (nileblack) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java:145]: Just want record detail data? what's best practice to do that?

**Comment by nileblack** [comment]
> @nileblack Are you still working on this?

Sorry, I'd left a few days. I'm back now.

**Comment by zifeihan** [comment]
> > @nileblack Are you still working on this?
> 
> Sorry, I'd left a few days. I'm back now.

Hello, I think you need to catch ʻException` in each kafka handler, because the kafka handler is driven by the submit method of the thread pool. In this case, an exception occurs. By default, no exception information will be printed.

**Review by wu-sheng — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
You need to fix conflicts.

**Comment by wu-sheng** [comment]
Merge master into your branch please.

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java (hunk 2 above)
Comment (wu-sheng) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java]: Logging data actually isn't useful.

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/JVMMetricsHandler.java [hunk 3]
```diff
@@ -67,8 +67,8 @@ public void handle(final ConsumerRecord<String, Bytes> record) {
             builder.getMetricsList().forEach(jvmMetric -> {
                 jvmSourceDispatcher.sendMetric(builder.getService(), builder.getServiceInstance(), jvmMetric);
             });
-        } catch (InvalidProtocolBufferException e) {
-            log.error("", e);
+        } catch (Exception e) {
+            log.error("handle record failed, record:{}", record.toString(), e);
```
Comment (wu-sheng) [inline oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/JVMMetricsHandler.java:71]: Logging error should be enough.

Comment (nileblack) [reply oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/JVMMetricsHandler.java]: done

**Review by wu-sheng — COMMENTED** [review]
`CHANGES.md` should be updated.

File: CHANGES.md [hunk 4]
```diff
@@ -26,6 +26,7 @@ Release Notes.
 * Improve Kubernetes service registry for ALS analysis.
 * Add health checker for cluster management
 * Improve the queryable tags generation. Remove the duplicated tags to reduce the storage payload.
+* Catch All Execption in `KafkaHandler`.
```
Comment (wu-sheng) [inline CHANGES.md:29]:
```suggestion
* Fix the threads of the Kafka fetcher exit if some unexpected exceptions happen.
```

**Review by wu-sheng — APPROVED** [review]
@zifeihan @dmsolr Recheck please.

**Review by wu-sheng — APPROVED** [review]

**Comment by zifeihan** [comment]
> @zifeihan @dmsolr Recheck please.

I think it is ok.

**Review by dmsolr — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
Welcome to be the 280th contributor to this repo. :) Congrats.

**Pull request merged**